packages/legend-query-builder/src/stores/explorer/QueryBuilderPropertySearchState.ts

```diff
@@ -41,45 +45,103 @@ import {
   QueryBuilderExplorerTreeSubTypeNodeData,
 } from './QueryBuilderExplorerState.js';
 import type { QueryBuilderState } from '../QueryBuilderState.js';
+import { Fuse, TextSearchAdvancedConfigState } from '@finos/legend-art';
 
 export class QueryBuilderPropertySearchState {
   queryBuilderState: QueryBuilderState;
   // TODO: Check if we could clean this up as this seems quite complicated and its purpose is not clear to me
   // See https://github.com/finos/legend-studio/pull/1212
-  allMappedPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+  mappedPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
   searchedMappedPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+
   isSearchPanelOpen = false;
+  isSearchConfigOpen = false;
   isSearchPanelHidden = false;
+  isOverSearchLimit = false;
+
+  textSearchState: TextSearchAdvancedConfigState;
+
   searchText = '';
+  searchState = ActionState.create().pass();
+  searchEngine: Fuse<QueryBuilderExplorerTreeNodeData>;
+
   filterByMultiple: boolean;
+
   typeFilters: QUERY_BUILDER_PROPERTY_SEARCH_TYPE[];
 
   constructor(queryBuilderState: QueryBuilderState) {
     makeAutoObservable(this, {
       queryBuilderState: false,
       searchedMappedPropertyNodes: observable,
       isSearchPanelOpen: observable,
+      isSearchConfigOpen: observable,
+      isOverSearchLimit: observable,
       isSearchPanelHidden: observable,
       searchText: observable,
+      textSearchState: observable,
+      searchEngine: observable,
       filteredPropertyNodes: computed,
       setSearchText: action,
+      setIsOverSearchLimit: action,
       setSearchedMappedPropertyNodes: action,
       setIsSearchPanelOpen: action,
+      setisSearchConfigOpen: action,
       setIsSearchPanelHidden: action,
-      refreshPropertyState: action,
+      resetPropertyState: action,
       setFilterByMultiple: action,
       toggleTypeFilter: action,
     });
 
     this.queryBuilderState = queryBuilderState;
     this.filterByMultiple = true;
+
+    this.textSearchState = new TextSearchAdvancedConfigState(
+      this.search.bind(this),
+    );
+
     this.typeFilters = [
       QUERY_BUILDER_PROPERTY_SEARCH_TYPE.CLASS,
       QUERY_BUILDER_PROPERTY_SEARCH_TYPE.STRING,
       QUERY_BUILDER_PROPERTY_SEARCH_TYPE.BOOLEAN,
       QUERY_BUILDER_PROPERTY_SEARCH_TYPE.NUMBER,
       QUERY_BUILDER_PROPERTY_SEARCH_TYPE.DATE,
     ];
+
+    this.searchEngine = new Fuse(this.mappedPropertyNodes);
+  }
+
+  setFuse(): void {
```

Comment:
We should almost never use name of the library in method, implementation can change. Also, I think it's perfectly fine to do inline this instead of having a separate method